aws-android-sdk-appsync/src/main/java/com/amazonaws/mobileconnectors/appsync/SubscriptionAuthorizer.java

```diff
@@ -63,58 +69,45 @@ JSONObject getAuthorizationDetails(boolean connectionFlag,
         // Get the Auth Mode from configuration json
         String authMode;
         try {
-            authMode = awsConfiguration.optJsonObject("AppSync").getString("AuthMode");
+            authMode = mAwsConfiguration.optJsonObject("AppSync").getString("AuthMode");
         } catch (JSONException e) {
             throw new RuntimeException("Failed to read AuthMode from awsconfiguration.json", e);
         }
 
         // Construct the Json based on the Auth Mode
         switch (authMode) {
             case "API_KEY" :
-                return getAuthorizationDetailsForApiKey(awsConfiguration);
+                return getAuthorizationDetailsForApiKey();
             case "AWS_IAM" :
-                return getAuthorizationDetailsForIAM(connectionFlag, awsConfiguration,
-                        subscription, applicationContext);
+                return getAuthorizationDetailsForIAM(connectionFlag, subscription);
             case "AMAZON_COGNITO_USER_POOLS" :
-                return getAuthorizationDetailsForUserpools(awsConfiguration, applicationContext);
+                return getAuthorizationDetailsForUserpools();
             case "OPENID_CONNECT" :
-                return getAuthorizationDetailsForOidc(awsConfiguration, mOidcAuthProvider);
+                return getAuthorizationDetailsForOidc();
```

Comment:
Changing these methods to be non-static since they are private and will only be used in instantiated instances of the `SubscriptionAuthorizer`. This way they can access whatever variables they need in the instance.